frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js

```diff
@@ -100,27 +101,34 @@ export class CustomHomePage extends Component {
     params: {
       variance_application_status_code: [],
       page: Strings.DEFAULT_PAGE,
-      per_page: 5,
+      per_page: Strings.DEFAULT_PER_PAGE,
       major: "",
       region: [],
       compliance_code: [],
-      issue_date_min: "",
-      issue_date_max: "",
-      expiry_date_max: "",
-      expiry_date_min: "",
+      issue_date_after: "",
+      issue_date_before: "",
+      expiry_date_before: "",
+      expiry_date_after: "",
       search: "",
     },
   };
 
   componentDidMount() {
     const params = this.props.location.search;
+    const parsedParams = queryString.parse(params);
+    const {
+      page = this.state.params.page,
+      per_page = this.state.params.per_page,
+      type = this.state.params.type,
+    } = parsedParams;
     if (params) {
       this.renderDataFromURL(params);
     } else {
       this.props.history.push(
         router.VARIANCE_DASHBOARD.dynamicRoute({
-          page: String.DEFAULT_PAGE,
-          per_page: String.DEFAULT_PER_PAGE,
+          page,
+          per_page,
+          type,
```

Comment:
Am I missing something? It looks like `page`, `per_page`, and `type` are pulled out of `parsedParams`, which is created by parsing `params`. So, if `params` doesn't exist (which could cause this block to run), then `page`, `per_page`, and `type` would always be the default values (`this.state.params.whatever`).